src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -465,6 +471,37 @@ private string GetValue(NodeModel node)
             return data;
         }
 
+        private string GenerateWatchData(List<ProtoCore.Mirror.MirrorData> list, int t)
+        {
+            var tab = new string('\t', t);
+            var format = "{0}[{1}] {2}{3}";
+            var data = new StringBuilder(list.Count == 0 ? "Empty List" : "List: \n");
+            var i = 0;
+
+            foreach (var mirrorData in list)
+            {
+                if (mirrorData.IsCollection)
+                {
+                    data.Append(string.Format(format, tab, i++, GenerateWatchData(mirrorData.GetElements(), ++t), ""));
+                }
+                else if(mirrorData.Data != null)
+                {
+                    double number;
+                    data.Append(double.TryParse(mirrorData.StringData, out number)
+                        ? string.Format(format, tab, i++, number.ToString("F4"), "\n")
+                        : string.Format(format, tab, i++, mirrorData.Data, "\n"));
+                }
+                else
+                {
+                    data.Append(mirrorData.Class == null 
+                        ? string.Format(format, tab, i++, "null", "\n")
+                        : string.Format(format, tab, i++, mirrorData.Class.ClassName, "\n"));
+                }
+            }
+
+            return data.ToString();
+        }
+
```

Comment:
This will be fine if we only display indented string data on client (without the capability to collapse a given level). I'm just being curious, is there a potential requirement in the future that may require us to display 'tree view' on the client? Also, how relevant is a suggestion to return JSON data here? :)